data-model/src/main/java/io/micronaut/data/model/query/builder/AbstractSqlLikeQueryBuilder.java

```diff
@@ -749,6 +753,11 @@ private void appendPropertyProjection(StringBuilder sb, QueryPropertyPath proper
         boolean escape = propertyPath.shouldEscape();
         NamingStrategy namingStrategy = propertyPath.getNamingStrategy();
         boolean[] needsTrimming = {false};
+        PersistentEntity entity = propertyPath.getProperty().getOwner();
+        if (entity.isJsonView()) {
```

Comment:
I have tried to implement this, but for some reason Oracle jdbc driver is not doing field retrieval like for regular tables (for string returns getSQLXML, for Double throws an exception) so I think we cannot support projections against views currently.